src/modules/dashboard/components/ColonyActions/ColonyActions.tsx

```diff
@@ -173,21 +174,37 @@ const ColonyActions = ({
   ]);
 
   /* Needs to be tested when all action types are wirde up & reflected in the list */
-  const filteredActions = useMemo(
-    () =>
-      !ethDomainId
-        ? actions
-        : actions.filter(
-            (action) =>
-              Number(action.fromDomain) === ethDomainId ||
-              /* when no specific domain in the action it is displayed in Root */
-              (ethDomainId === 1 && action.fromDomain === undefined) ||
-              /* when transfering funds the list shows both sender & recipient */
-              (action.actionType === ColonyActionTypes.MoveFunds &&
-                Number(action.toDomain) === ethDomainId),
-          ),
-    [ethDomainId, actions],
-  );
+  const filteredActions = useMemo(() => {
+    let filterActions = actions;
+    filterActions = !ethDomainId
+      ? actions
+      : actions.filter(
+          (action) =>
+            Number(action.fromDomain) === ethDomainId ||
+            /* when no specific domain in the action it is displayed in Root */
+            (ethDomainId === 1 && action.fromDomain === undefined) ||
+            /* when transfering funds the list shows both sender & recipient */
+            (action.actionType === ColonyActionTypes.MoveFunds &&
+              Number(action.toDomain) === ethDomainId),
+        );
+
+    /* filter out duplicate action items on passed motions */
+    if (motions && motions.motions?.length > 0) {
+      const motionExtensionAddresses =
+        motions.motions
+          .filter((motion) => motion.extensionAddress)
+          .map((motion) => createAddress(motion.extensionAddress)) || [];
+      if (motionExtensionAddresses.length > 0) {
+        filterActions = actions.filter((action) => {
+          return !motionExtensionAddresses.includes(
+            createAddress(action.initiator),
+          );
+        });
```

Comment:
so when you get rid of let, you can just put return here in if

Reply:
Thanks you, that helps!